### PR TITLE
Add description about the stage prefix for symfony

### PR DESCRIPTION
**Pull request by jenschude**
It can be quite tricky to get correctly handle the stage prefix in Symfony.

### CODE REVIEW

**Comment by Nyholm** [comment]
Thank you for this PR. 

I see that the `ROUTE_BASE_URL` variable is the switch to turn this on/off. Could we do it in a bit more automatic way? Ie, not use custom environment variables

**Comment by jenschude** [comment]
Yeah tried it:

https://github.com/brefphp/bref/compare/master...jenschude:set_script_name_env

Issue is still, that it's not reliable possible to get the information from where the request originates and what the real calling URI was.

**Comment by Nyholm** [comment]
If #542 gets merged, would that help? I think it would, right?

**Comment by jenschude** [comment]
Depends. The bref.dev service for example acts like a normal client and initially didn‘t set a Forwarded Header. Which made it impossible to read the called URI

File: docs/frameworks/symfony.md [hunk 1]
```diff
@@ -116,3 +116,30 @@ Globals:
...
+}
+```
+
+This assumes that your function has a environment variable named `ROUTE_BASE_URL` which will hold the stage prefix like `/dev`.
+To automatically set this variable add this snippet to your serverless.yaml
+
+```yaml
+provider:
+  environment:
+    ROUTE_BASE_URL: '/${self:custom.stage}'
```
Comment (xavismeh) [inline docs/frameworks/symfony.md:142]: Using `/${self:provider.stage, 'dev'}` will allow to do not determine an extra custom parameter reusing an already defined one

**Comment by Nemo64** [comment]
Since the http api is now generally available, wouldn't it make more sense to slowly change everything to the http api which does not need a prefix?
There are few cases where you need the rest api and if you need them you are probably very deep into it.

**Comment by jenschude** [comment]
When the pull request was initially opened the default for serverless was to have stage prefixes. This can be circumvented now. But even if deployed with prefixes one has to know some internals of Symfony how to correctly setup the environment so that route prefixes are correctly identified especially as Symfony defaults to generate absolute URLs.

**Comment by mnapoli** [comment]
Since #309 is fixed in #794 and we switch to API Gateway v2, this problem will disappear. Thanks for the contribution anyway!

**Pull request closed without merging**